lib/redis/namespace.rb

```diff
@@ -372,7 +372,14 @@ def method_missing(command, *args, &block)
                "passthrough has been deprecated and will be removed in " +
                "redis-namespace 2.0 (at #{call_site})")
         end
-        @redis.send(command, *args, &block)
+
+        if @redis.class.name == "ConnectionPool"
+          @redis.with do |pool_connection|
```

Comment:
I'm not familiar with connection pool: is there a way to get the `pool_connection` without using `with`?

i.e. we could do something like:

```ruby
redis_connection = if @redis.class.name == "ConnectionPool"
  @redis.connection_pool
else
  @redis
end
```

And then have a single `send` call?

Reply:
Figured, makes sense!